tests/fixtures/expect.go

```diff
@@ -34,38 +37,117 @@ type Expect struct {
+
 	for _, r := range resources {
-		result := e.kubeClient.CoreV1().RESTClient().Get().
-			Timeout(defaultTimeout).
-			Namespace(TargetNamespace).
-			Resource(resourceType).
-			Name(r).
-			Do(ctx)
-		if result.Error() != nil {
-			e.t.Fatalf("Resource %s does not exist", r)
+		if !e.doesExist(apiVersion, resourceType, r) {
+			e.t.Fatalf("Resource %s/%s does not exist", resourceType, r)
 		}
+		e.t.Logf("Resource %s/%s exists", resourceType, r)
 	}
 
 	return e
 }
 
-// check that resources are deleted
-func (e *Expect) ResourcesDontExist(resourceType string, resources []string) *Expect {
+// check that resources have been deleted or never existed
+func (e *Expect) ResourcesDontExist(apiVersion, resourceType string, resources []string) *Expect {
 
 	e.t.Helper()
+	e.t.Log("Verifying that resources no longer exist..")
+
 	for _, r := range resources {
-		if !e.isDeleted(resourceType, r) {
-			e.t.Fatalf("Resource %s not deleted", r)
+		if !e.doesNotExist(apiVersion, resourceType, r) {
+			e.t.Fatalf("Resource %s/%s not deleted", resourceType, r)
+		}
+		e.t.Logf("Resource %s/%s doesn't exist", resourceType, r)
+	}
+
+	return e
+}
+
+// verify value of resource spec to determine if change occurred or not as expected
+func (e *Expect) VerifyResourceState(apiVersion, resourceType, resource, field, key string, value interface{}) *Expect {
+
+	e.t.Helper()
+	e.t.Log("Verifying resource state is as expected..")
+
+	var (
+		err  error
+		body []byte
+	)
+
+	if !e.doesExist(apiVersion, resourceType, resource) {
+		e.t.Fatalf("Resource %s/%s does not exist", resourceType, resource)
+	}
+
+	ctx := context.Background()
+
+	if apiVersion == "v1" {
+		body, err = e.kubeClient.CoreV1().RESTClient().
+			Get().
+			Namespace(TargetNamespace).
+			Resource(resourceType).
+			Name(resource).
+			DoRaw(ctx)
+	} else {
+		body, err = e.kubeClient.CoreV1().RESTClient().Get().AbsPath(
+			fmt.Sprintf("/apis/%s/namespaces/%s/%s/%s",
+				apiVersion,
+				TargetNamespace,
+				resourceType,
+				resource)).DoRaw(ctx)
+	}
+
+	if err != nil {
+		e.t.Fatalf("Failed to verify resource %s/%s state", resourceType, resource)
+	}
+
+	object := make(map[string]interface{})
+	err = yaml.Unmarshal(body, object)
+	if err != nil {
+		e.t.Fatalf("Failed to verify resource %s/%s state", resourceType, resource)
+	}
+
+	for k, v := range object[field].(map[interface{}]interface{}) {
+		if k == key {
+			if v != value {
+				e.t.Fatalf("Resource %s/%s state is not as expected", resourceType, resource)
+			}
 		}
 	}
 
+	e.t.Log("Resource state is as expected")
+
+	return e
+}
+
+// verify that GitSync's commitStatus is as expected:
+// synced = true
+// hash is equal to the last commit to the Git server
+func (e *Expect) CheckCommitStatus() *Expect {
+
+	e.t.Helper()
+	e.t.Log("Verifying GitSync's commitStatus is as expected..")
+
+	// gitSync commit status will be nil if reconciliation has not finished
+	commitStatus, err := e.getCommitStatus()
```

Comment:
I see there's a timeout in `getCommitStatus()` to allow some time for the `CommitStatus != nil`. I'm kind of wondering if the timeout should be over here, though. Consider that initially a GitSync will have no CommitStatus, then after the first reconciliation, there's a CommitStatus. Then something changes in the synced resources, and the CommitStatus gets updated to the new one. If we call `CheckCommitStatus()` after that last time, it will immediately return from `getCommitStatus()` because there is a CommitStatus (the old one maybe), and then here it may find that the hash doesn't match. Let me know if I have that wrong.

Reply:
This is a good callout, I need to revisit this method tomorrow. Initially when I was checking for `CommitStatus` I was unsure why I kept getting a nil pointer and thats why I included the timeout there at the time. Would timing out on the condition that the `hash` is as we expected better? This assumes `Synced == true` as well.

Reply:
Sure thing, I'll come up with a definitive answer tomorrow.